pkg/sql/exec_util.go

```diff
@@ -2929,11 +2929,14 @@ func (m *sessionDataMutator) bufferParamStatusUpdate(param string, status string
 
 // SetApplicationName sets the application name.
 func (m *sessionDataMutator) SetApplicationName(appName string) {
+	oldName := m.data.ApplicationName
 	m.data.ApplicationName = appName
 	if m.onApplicationNameChange != nil {
 		m.onApplicationNameChange(appName)
 	}
-	m.bufferParamStatusUpdate("application_name", appName)
+	if oldName != appName {
+		m.bufferParamStatusUpdate("application_name", appName)
```

Comment:
i wonder if we should make `bufferParamStatusUpdate` always take in the old value instead and do it all in one place.

Reply:
hmm nvm, for `is_superuser` we always have to send that over.